frappe/model/delete_doc.py

```diff
@@ -321,14 +321,11 @@ def check_if_doc_is_dynamically_linked(doc, method="Delete"):
 				(doc.doctype, doc.name),
 				as_dict=True,
 			):
-
-				if (method == "Delete" and refdoc.docstatus < 2) or (
-					method == "Cancel" and refdoc.docstatus == 1
+				# linked to an non-cancelled doc when deleting
+				# or linked to a submitted doc when cancelling
+				if (method == "Delete" and not DocStatus(refdoc.docstatus).is_cancelled()) or (
```

Comment:
though the logic is correct *for now* ..`!= 2` will include any further docstatuses if introduced in future

any higher docstatus than 2 should be deletable?

Reply:
Are you serious? 😅 Even the comment (originally by Rushabh) says "non-cancelled". When was the last time we introduced a new docstatus?

Reply:
yep - i had my doubts regarding my own comment but thought why not :man_shrugging: :P

it's correct anyway - will merge